cwl/src/main/scala/wdl4s/cwl/Workflow.scala

```diff
@@ -76,21 +74,23 @@ case class Workflow(
 
           val wdlType = cwlTypeToWdlType(output.`type`.flatMap(_.select[CwlType]).get)
 
-          def lookupOutputSource(source: String): ErrorOr[OutputPort] =
-            graphFromSteps.
-              flatMap(_.outputPorts).
-              find(_.name == source).
-              toValidNel(s"unable to find upstream port corresponding to $source")
+          def lookupOutputSource(outputId: WorkflowOutputId): ErrorOr[OutputPort] = {
+            (for {
+              call <- graphFromSteps.collectFirst { case callNode: CallNode if callNode.name == outputId.stepId => callNode }
+              output <- call.outputPorts.find(_.name == outputId.outputId)
+            } yield output).toValidNel(s"unable to find upstream port corresponding to ${outputId.stepId}/${outputId.outputId}")
+          }
 
-          lookupOutputSource(output.outputSource.flatMap(_.select[String]).get).
+          lookupOutputSource(WorkflowOutputId(output.outputSource.flatMap(_.select[String]).get)).
             map(PortBasedGraphOutputNode(output.id, wdlType, _))
       }.map(_.toSet)
 
     graphFromOutputs.flatMap(outputs => Graph.validateAndConstruct(graphFromSteps ++ graphFromInputs ++ outputs))
   }
 
   def womDefinition: ErrorOr[WorkflowDefinition] = {
-    val name: String = "workflow Id"
+    // TODO: need to find a way to get a meaningful name here
+    val name: String = "MyCwlWorkflow"
```

Comment:
assuming this needs to be a stable name too (doesn't change for a given workflow source)?

Reply:
Hmm yes I think you're right if we want things like restart to work in Cromwell